misc/userscripts/password_fill

```diff
@@ -0,0 +1,276 @@
+}
+
+choose_entry_zenity_radio() {
+    zenity_helper() {
+        awk '{ print $0 ; print $0 }'                   \
+        | zenity --list --radiolist                     \
+                 --title "Qutebrowser password fill"    \
+                 --text "Pick the password entry:"      \
+                 --column " " --column "Name"
+    }
+    MENU_COMMAND=( zenity_helper )
+    choose_entry_menu || true
+}
+
+# =======================================================
+# backend: PASS
+
+# configuration options:
+match_filename=1 # whether allowing entry match by filepath
+match_line=1     # whether allowing entry match by URL-Pattern in file
+match_line_pattern='^url: .*' # applied using grep -iE
+user_pattern='^(user|username): '
+
+GPG_OPTS=( "--quiet" "--yes" "--compress-algo=none" "--no-encrypt-to" )
+GPG="gpg"
+export GPG_TTY="${GPG_TTY:-$(tty 2>/dev/null)}"
+which gpg2 &>/dev/null && GPG="gpg2"
+[[ -n $GPG_AGENT_INFO || $GPG == "gpg2" ]] && GPG_OPTS+=( "--batch" "--use-agent" )
+
+pass_backend() {
+    init() {
+        PREFIX="${PASSWORD_STORE_DIR:-$HOME/.password-store}"
+        if ! [ -d "$PREFIX" ] ; then
+            die "Can not open password store dir »$PREFIX«"
+        fi
+    }
+    query_entries() {
+        local url="$1"
+
+        if ((match_line)) ; then
+            # add entries with matching URL-tag
+            while read -r -d "" passfile ; do
+                if $GPG "${GPG_OPTS}" -d "$passfile" \
+                     | grep --max-count=1 -iE "${match_line_pattern}${url}" > /dev/null
+                then
+                    passfile="${passfile#$PREFIX}"
+                    passfile="${passfile#/}"
+                    files+=( "${passfile%.gpg}" )
+                fi
+            done < <(find -L "$PREFIX" -iname '*.gpg' -print0)
+        fi
+        if ((match_filename)) ; then
+            # add entries wth matching filepath
+            while read -r passfile ; do
+                passfile="${passfile#$PREFIX}"
+                passfile="${passfile#/}"
+                files+=( "${passfile%.gpg}" )
+            done < <(find -L "$PREFIX" -iname '*.gpg' | grep "$url")
+        fi
+    }
+    open_entry() {
+        local path="$PREFIX/${1}.gpg"
+        password=""
+        local firstline=1
+        while read -r line ; do
+            if ((firstline)) ; then
+                password="$line"
+                firstline=0
+            else
+                if [[ $line =~ $user_pattern ]] ; then
+                    # remove the matching prefix "user: " from the beginning of the line
+                    username=${line#${BASH_REMATCH[0]}}
+                    break
+                fi
+            fi
+        done < <($GPG "${GPG_OPTS}" -d "$path" )
+    }
+}
+# =======================================================
+
+# load some sane default backend
+reset_backend
+pass_backend
+# load configuration
+QUTE_CONFIG_DIR=${QUTE_CONFIG_DIR:-${XDG_CONFIG_HOME:-$HOME/.config}/qutebrowser/}
```

Comment:
Adding `QUTE_CONFIG_DIR` (and some other `*_DIR` things) to the userscript variables for qutebrowser sounds like a good idea.